examples/object_detection_3d/object_detection_3d/evaluator.py

```diff
@@ -0,0 +1,374 @@
+# Copyright 2021-2023 Kolena Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from dataclasses import dataclass
+from enum import IntEnum
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Set
+from typing import Tuple
+
+import numpy as np
+
+from .kitti_eval import kitti_eval
+from .workflow import GroundTruth
+from .workflow import Inference
+from .workflow import TestCase
+from .workflow import TestSample
+from kolena.workflow import Evaluator
+from kolena.workflow import EvaluatorConfiguration
+from kolena.workflow import MetricsTestCase as BaseMetricsTestCase
+from kolena.workflow import MetricsTestSample as BaseMetricsTestSample
+from kolena.workflow.annotation import BoundingBox
+from kolena.workflow.annotation import BoundingBox3D
+from kolena.workflow.plot import Curve
+from kolena.workflow.plot import CurvePlot
+from kolena.workflow.plot import Plot
+
+
+VALID_LABELS = ["Car", "Pedestrian", "Cyclist"]
+
+
+class KITTIDifficulty(IntEnum):
+    """
+    SOURCE: http://www.cvlibs.net/datasets/kitti/eval_object.php?obj_benchmark=3d
+
+    For cars we require an 3D bounding box overlap of 70%,
+    while for pedestrians and cyclists we require a 3D bounding box overlap of 50%.
+
+    Difficulties are defined as follows:
+
+    Easy: Min. bounding box height: 40 Px, Max. occlusion level: Fully visible, Max. truncation: 15 %
+    Moderate: Min. bounding box height: 25 Px, Max. occlusion level: Partly occluded, Max. truncation: 30 %
+    Hard: Min. bounding box height: 25 Px, Max. occlusion level: Difficult to see, Max. truncation: 50 %
+    """
+
+    EASY = 0
+    MODERATE = 1
+    HARD = 2
+
+
+@dataclass(frozen=True)
+class KITTI3DConfig(EvaluatorConfiguration):
+    difficulty: KITTIDifficulty
+
+    def display_name(self):
+        return f"KITTI Difficulty ({self.name()})"
+
+    def name(self):
+        if self.difficulty == KITTIDifficulty.EASY:
+            name = "easy"
+        elif self.difficulty == KITTIDifficulty.MODERATE:
+            name = "moderate"
+        else:
+            name = "hard"
+        return name
+
+
+@dataclass(frozen=True)
+class MetricsTestSample(BaseMetricsTestSample):
+    n_gts: int
+    n_infs: int
+    n_valid_gts: int
```

Comment:
Is adding TP/FP/FN bboxes an intended follow-up?

Reply:
Can do this as a follow-up, or prioritize including it in the first delivery. What do you think?

Reply:
Sounds good to me